src/WindowManager.vala

```diff
@@ -722,18 +722,19 @@ namespace Gala {
                 out x, out y);
         }
 
-        private void toggle_dim_parent (Meta.Window window) {
+        private void dim_parent_window (Meta.Window window, bool dim) {
             if (window.window_type == Meta.WindowType.MODAL_DIALOG) {
                 var ancestor = window.find_root_ancestor ();
                 if (ancestor != null && ancestor != window) {
                     var win = (Meta.WindowActor) ancestor.get_compositor_private ();
-                    if (win.has_effects ()) {
-                        win.clear_effects ();
-                    } else {
+                    // Can't rely on win.has_effects since other effects could be applied
+                    if (dim) {
                         var dark_effect = new Clutter.BrightnessContrastEffect ();
                         dark_effect.set_brightness (-0.4f);
 
                         win.add_effect (dark_effect);
+                    } else {
+                        win.clear_effects ();
```

Comment:
A comment mentioned above says other effects could be applied so wouldn't this clear not just the intended dim effect but also any other effects not intended to be removed?